scipy/stats/_survival.py

```diff
@@ -374,3 +391,138 @@ def _ecdf_right_censored(sample):
     sf = np.cumprod((n - d) / n)
     cdf = 1 - sf
     return t, cdf, sf, n, d
+
+
+@dataclass
+class LogRankResult:
+    statistic: np.ndarray
+    pvalue: np.ndarray
+
+
+def log_rank(
```

Comment:
```suggestion
def logrank(
```
It's commonly written as a single word (e.g. everywhere in this function's documentation), and this is easier to type.